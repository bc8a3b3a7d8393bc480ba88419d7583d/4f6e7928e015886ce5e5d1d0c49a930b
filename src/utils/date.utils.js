import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
