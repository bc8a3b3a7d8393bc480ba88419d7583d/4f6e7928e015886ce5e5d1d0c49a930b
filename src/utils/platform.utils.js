import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
