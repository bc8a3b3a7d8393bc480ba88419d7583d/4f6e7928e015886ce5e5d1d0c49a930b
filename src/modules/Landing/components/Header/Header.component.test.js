import { testSnapshots } from '../../../../utils/test.utils';

import Component from './Header.component';

describe('Modules/Landing - Header', () => {
  testSnapshots(Component, [
    {
      description: 'basic render',
      props: {
        languageCode: 'en',
        languageName: 'English',
        countryCode: 'VN',
        countryName: 'Viet Nam',
      },
    },
  ]);
});
