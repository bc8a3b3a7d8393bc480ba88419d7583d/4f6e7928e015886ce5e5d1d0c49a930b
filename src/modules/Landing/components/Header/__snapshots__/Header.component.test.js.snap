// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules/Landing - Header snapshots basic render 1`] = `
<View
  style={Object {}}
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/app/logo-white.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 66,
        "marginTop": 50,
        "width": 80,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Oswald-Regular",
        "fontSize": 20,
        "marginVertical": 30,
        "textAlign": "center",
      }
    }
  >
    landing.header.welcome
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Oswald-Regular",
        "fontSize": 12,
        "textAlign": "center",
      }
    }
  >
    landing.header.detectedLanguage
  </Text>
  <LanguageItem
    containerStyle={
      Object {
        "marginVertical": 15,
      }
    }
    countryCode="VN"
    countryName="Viet Nam"
    languageCode="en"
    languageName="English"
    onPress={[Function]}
    selected={true}
  />
</View>
`;
